modules/planning/tasks/deciders/open_space_fallback_decider.cc

```diff
@@ -55,63 +56,40 @@ Status OpenSpaceFallbackDecider::Process(Frame* frame) {
-
     const auto collision_point =
         ptr_fallback_trajectory_pair->first[first_collision_idx];
-    auto previous_point = ptr_fallback_trajectory_pair->first[0];
-
-    double relative_collision_point_x =
-        collision_point.path_point().x() - previous_point.path_point().x();
-    double relative_collision_point_y =
-        collision_point.path_point().y() - previous_point.path_point().y();
+    auto previous_point =
+        ptr_fallback_trajectory_pair->first[current_idx];
     double relative_collision_distance =
-        std::sqrt(relative_collision_point_x * relative_collision_point_x +
-                  relative_collision_point_y * relative_collision_point_y);
-    double stopping_distance =
-        std::min(relative_collision_distance,
-                 config_.open_space_fallback_decider_config()
-                     .open_space_fall_back_stop_distance());
-
-    if (stopping_distance > 0.0) {
-      // the accelerate = -v0^2 / (2*s), where s is slowing down distance
-      size_t temp_horizon =
-          frame_->open_space_info().fallback_trajectory().first.NumOfPoints();
-
-      const double accelerate =
-          -previous_point.v() * previous_point.v() / (2.0 * stopping_distance);
-
-      for (size_t i = 1; i < temp_horizon; ++i) {
-        double temp_relative_time =
-            ptr_fallback_trajectory_pair->first[i].relative_time() -
-            previous_point.relative_time();
-        double temp_v = previous_point.v() + accelerate * temp_relative_time;
-
-        if (temp_v <= 0.0) {
-          ptr_fallback_trajectory_pair->first[i].mutable_path_point()->set_x(
-              previous_point.path_point().x());
-          ptr_fallback_trajectory_pair->first[i].mutable_path_point()->set_y(
-              previous_point.path_point().y());
-          ptr_fallback_trajectory_pair->first[i].set_v(0.0);
-          ptr_fallback_trajectory_pair->first[i].set_a(0.0);
-        } else {
-          ptr_fallback_trajectory_pair->first[i].set_v(temp_v);
-          ptr_fallback_trajectory_pair->first[i].set_a(accelerate);
-        }
-
-        previous_point = ptr_fallback_trajectory_pair->first[i];
-      }
-    } else {
-      // if the stop at first idx
-      size_t temp_horizon =
-          frame_->open_space_info().fallback_trajectory().first.NumOfPoints();
-      const auto first_path_point =
-          ptr_fallback_trajectory_pair->first[0].path_point();
-      for (size_t i = 0; i < temp_horizon; ++i) {
-        ptr_fallback_trajectory_pair->first[i].set_v(0.0);
+        collision_point.path_point().s() - previous_point.path_point().s();
+    double stopping_distance = relative_collision_distance;
+    // the accelerate = -v0^2 / (2*s), where s is slowing down distance
+    size_t temp_horizon =
+        frame_->open_space_info().fallback_trajectory().first.NumOfPoints();
+    const double accelerate =
+        -std::abs(previous_point.v()) * previous_point.v()
```

Comment:
why std::abs here

Reply:
> why std::abs here

The accelerate is the -sign(current speed), so it is - |v| * v / (2 * s)